src/components/App.jsx

```diff
@@ -1,123 +1,66 @@
-        },
-        {
-          id: "5",
-          name: "Bohemian Rhapsody",
-          category: "Drama, Biography, Music",
-          year: "2003",
-          image:
-            "https://es.web.img3.acsta.net/pictures/18/05/21/12/50/1518397.jpg",
-        },
-        {
-          id: "6",
-          name: "Bohemian Rhapsody",
-          category: "Drama, Biography, Music",
-          year: "2003",
-          image:
-            "https://es.web.img3.acsta.net/pictures/18/05/21/12/50/1518397.jpg",
-        },
-        {
-          id: "7",
-          name: "Bohemian Rhapsody",
-          category: "Drama, Biography, Music",
-          year: "2003",
-          image:
-            "https://es.web.img3.acsta.net/pictures/18/05/21/12/50/1518397.jpg",
-        },
-        {
-          id: "8",
-          name: "Bohemian Rhapsody",
-          category: "Drama, Biography, Music",
-          year: "2003",
-          image:
-            "https://es.web.img3.acsta.net/pictures/18/05/21/12/50/1518397.jpg",
-        },
-        {
-          id: "9",
-          name: "Bohemian Rhapsody",
-          category: "Drama, Biography, Music",
-          year: "2003",
-          image:
-            "https://es.web.img3.acsta.net/pictures/18/05/21/12/50/1518397.jpg",
-        },
-      ],
-      app: { title: "netflix", subtitle: "roulette" },
-    };
-  }
+export const App = () => {
+  const userDetailContext = createContext(null);
+
+  //user mock to implement context
+  const [userDetails] = useState({
+    name: "Andres",
+    id: "000033",
+  });
+
+  const app = { title: "netflix", subtitle: "roulette" };
+
+  const [movies, setMovies] = useState(moviesData);
+  const [selectedMovie, setMovieSelected] = useState(null);
 
-  componentDidMount() {
-    setInterval(() => this.randomBestMovies(), 1000);
-  }
+  //custom hook to return movies count
+  const useTotalMovies = (movies) => {
+    const [moviesCount, setMoviesCount] = useState(0);
 
-  randomBestMovies() {
-    this.setState({
-      movies: [
-        {
-          id: "1",
-          name: "Pulp Fiction",
-          category: "Action & Adventure",
-          year: "2004",
-          image:
-            "https://es.web.img3.acsta.net/pictures/18/05/21/12/50/1518397.jpg",
-        },
-      ],
+    useEffect(() => {
+      setMoviesCount(movies.length);
     });
-  }
 
-  render() {
-    const { movies, app } = this.state;
-    return (
+    return moviesCount;
+  };
+
+  const movieSelectHandler = (event) => {
+    window.scrollTo(0, 0);
+    console.log("event here", event);
+    setMovieSelected(event);
+  };
+
+  const searchSelected = () => {
+    setMovieSelected(null);
+  };
+
+  return (
+    <userDetailContext.Provider value={userDetails}>
       <div className="container_app">
-        <Header app={app} />
+        {/* TODO: make component to show moviesCount */}
+        <span style={{ display: "none" }}>
+          Total movies: {useTotalMovies(moviesData)}
```

Comment:
What about using movies from useState as useTotalMovies param?. It is initialized with moviesData and later on it is possible for it to store server response for available movies.